sql/src/test/java/org/apache/apex/malhar/sql/EmbeddedKafka.java

```diff
@@ -0,0 +1,167 @@
+    KafkaConfig config = new KafkaConfig(props);
+    Time mock = new MockTime();
+    kafkaServer = TestUtils.createServer(config, mock);
+  }
+
+  public void stop()
+  {
+    kafkaServer.shutdown();
+    zkClient.close();
+    zkServer.shutdown();
+    cleanupDir();
+  }
+
+  private void cleanupDir()
+  {
+    try {
+      FileUtils.deleteDirectory(new File(KAFKA_PATH));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void createTopic(String topic)
+  {
+    AdminUtils.createTopic(zkUtils, topic, 1, 1, new Properties());
+    List<KafkaServer> servers = new ArrayList<KafkaServer>();
+    servers.add(kafkaServer);
+    TestUtils.waitUntilMetadataIsPropagated(scala.collection.JavaConversions.asScalaBuffer(servers), topic, 0, 30000);
+  }
+
+  public void publish(String topic, List<String> messages)
+  {
+    Properties producerProps = new Properties();
+    producerProps.setProperty("bootstrap.servers", BROKERHOST + ":" + BROKERPORT);
+    producerProps.setProperty("key.serializer","org.apache.kafka.common.serialization.IntegerSerializer");
+    producerProps.setProperty("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
+    KafkaProducer<Integer, byte[]> producer = new KafkaProducer<>(producerProps);
+
+    for (String message : messages) {
+      ProducerRecord<Integer, byte[]> data = new ProducerRecord<>(topic, message.getBytes(StandardCharsets.UTF_8));
+      producer.send(data);
+    }
+
+    producer.close();
+    List<KafkaServer> servers = new ArrayList<KafkaServer>();
+    servers.add(kafkaServer);
+    TestUtils.waitUntilMetadataIsPropagated(scala.collection.JavaConversions.asScalaBuffer(servers), topic, 0, 30000);
+  }
+
+  public List<String> consume(String topic, int timeout)
+  {
+    return consume(topic, timeout, true);
+  }
+
+  public List<String> consume(String topic, int timeout, boolean earliest)
+  {
+    Properties consumerProps = new Properties();
+    consumerProps.setProperty("bootstrap.servers", BROKERHOST + ":" + BROKERPORT);
+    consumerProps.setProperty("group.id", "group0");
+    consumerProps.setProperty("client.id", "consumer0");
+    consumerProps.setProperty("key.deserializer","org.apache.kafka.common.serialization.IntegerDeserializer");
+    consumerProps.setProperty("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+    consumerProps.put("auto.offset.reset", earliest ? "earliest" : "latest");  // to make sure the consumer starts from the beginning of the topic
+    KafkaConsumer<Integer, byte[]> consumer = new KafkaConsumer<>(consumerProps);
+    consumer.subscribe(Arrays.asList(topic));
+
+    List<String> messages = new ArrayList<>();
+
+    ConsumerRecords<Integer, byte[]> records = consumer.poll(timeout);
+    for (ConsumerRecord<Integer, byte[]> record : records) {
+      messages.add(new String(record.value()));
+    }
+
+    consumer.close();
```

Comment:
Use try-with-resource because close() may not be called if an exception is thrown before that.